### PR TITLE
Backups V2 Follow-up Fixes / Improvements

### PR DESCRIPTION
**Pull request by walmat**
Fixes APP-1297
Fixes APP-1050
Fixes APP-2002
Fixes APP-1649
Fixes APP-2120

## What changed (plus any additional context for devs)

This PR introduces a lot of stability changes to backups. I'll layout the important changes here.
- Removes the dependency on `UserData.json` since it quickly becomes muddy dealing with backup data and user data.
- Instead of adding to a backup file, we create a new backup file every time now to prevent potentially corrupting a backup file.
- Fixes the missing app portal when restoring / creating a wallet (or group)
- Fixes the portal which used to wrap the entire app and instead use a Zustand store to control whether or not it's shown
- Instead of relying on local wallet state to determine whether or not they have enabled the cloud backups, we now determine that based on if they have at least one backup.
- Adds `backupsStore` Zustand store which controls the synced backup data
- Prevents any backup actions (create/restore) if the local state isn't in sync with remote data
- Fixes bug with local password being saved as empty string
- Fixes bug where restoring to backup wouldn't navigate back to wallet screen
- Fixes bug where incorrect password indicator wasn't shown on restore screen
- Fixes issue where notifications + app icon unlock sheets would be stacked on new wallets
- Fixes native crash with syncing cloud-fs in-flight multiple times

## What to test
https://www.notion.so/rainbowdotme/Backups-V2-Fixes-QA-Hand-off-149b001b85b4800fbf72c0f347b992b0?showMoveTo=true&saveParent=true

### CODE REVIEW

File: .vscode/settings.json
Comment (walmat) [inline .vscode/settings.json:15]:
```suggestion
  ]
```

File: src/App.tsx
Comment (walmat) [inline src/App.tsx:86]: These are now no longer re-rendering the entire app when re-rendering

File: src/App.tsx
Comment (walmat) [inline src/App.tsx:87]: In charge of initial backup store sync on mount

File: src/App.tsx
Comment (walmat) [inline src/App.tsx:105]: we only really want to re-render the App when walletReady has changed.. this fixes that

File: src/components/backup/CloudBackupProvider.tsx
Comment (walmat) [inline src/components/backup/CloudBackupProvider.tsx:1]: this logic was moved into backupStore

File: src/components/backup/RestoreCloudStep.tsx
Comment (walmat) [inline src/components/backup/RestoreCloudStep.tsx:153]:
```suggestion
        backupFilename: filename,
```

File: src/components/secret-display/SecretDisplaySection.tsx
Comment (walmat) [inline src/components/secret-display/SecretDisplaySection.tsx:132]: coming from `undefined` -> `manual` if not already set to `cloud`

File: src/handlers/cloudBackup.ts
Comment (walmat) [inline src/handlers/cloudBackup.ts:109]: this is dependent on https://github.com/rainbow-me/react-native-cloud-fs/pull/20

File: src/components/backup/RestoreCloudStep.tsx
Comment (greg-schrammel) [inline src/components/backup/RestoreCloudStep.tsx:106]: isn't the `reset` useful here? https://reactnavigation.org/docs/6.x/navigation-actions/#reset

File: src/helpers/walletLoadingStates.ts
Comment (greg-schrammel) [inline src/helpers/walletLoadingStates.ts:8]: i18n?

**Comment by brunobar79** [comment]
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/@matthew/APP-1297/45fa243afba914fec263f35954f8274fd280a323.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/@matthew/APP-1297/45fa243afba914fec263f35954f8274fd280a323.ipa) for 45fa243afba914fec263f35954f8274fd280a323

**Comment by brunobar79** [comment]
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/@matthew/APP-1297/a59c53d807c8ff7c2165a1be0b38a8a0fb93930b.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/@matthew/APP-1297/a59c53d807c8ff7c2165a1be0b38a8a0fb93930b.ipa) for a59c53d807c8ff7c2165a1be0b38a8a0fb93930b

**Comment by brunobar79** [comment]
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/@matthew/APP-1297/43058f6ddb84144b88def6a0e9ea0a8a111f39bd.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/@matthew/APP-1297/43058f6ddb84144b88def6a0e9ea0a8a111f39bd.ipa) for 43058f6ddb84144b88def6a0e9ea0a8a111f39bd

**Comment by brunobar79** [comment]
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/@matthew/APP-1297/93c7d837469e9bc457c8c257c4d0daf6d8769e2d.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/@matthew/APP-1297/93c7d837469e9bc457c8c257c4d0daf6d8769e2d.ipa) for 93c7d837469e9bc457c8c257c4d0daf6d8769e2d

File: src/components/AbsolutePortal.tsx
Comment (greg-schrammel) [inline src/components/AbsolutePortal.tsx:27]: are you using this component?

File: src/screens/SettingsSheet/components/Backups/WalletsAndBackup.tsx
Comment (greg-schrammel) [inline src/screens/SettingsSheet/components/Backups/WalletsAndBackup.tsx:675]: ah here this file was collapsed mb
I don't understand how the navigator works really, in the network switcher pr I added this `<AbsolutePortalRoot />` to the root of the app right next to `<Routes />` and it works on most screens, but not like in the swaps thats kinda of a sheet route, in that we have another `<AbsolutePortalRoot />` specific for that route, but ideally we should have only one of this component for the whole app, or change the singleton store that controls it
doesn't having a single `<AbsolutePortalRoot />` with a big zIndex like that work? I don't think I tried that

File: src/components/floating-emojis/FloatingEmojis.tsx
Comment (derHowie) [inline src/components/floating-emojis/FloatingEmojis.tsx:71]: is this worth a discussion ticket?

File: src/screens/SettingsSheet/utils.ts
Comment (derHowie) [inline src/screens/SettingsSheet/utils.ts:48]: `acc.canBeBackedUp || isBackupEligible` --> should this be `acc.canBeBackedUp && isBackupEligible`

File: src/screens/SettingsSheet/utils.ts
Comment (derHowie) [inline src/screens/SettingsSheet/utils.ts:62]: same

File: src/hooks/useActiveRoute.ts
Comment (derHowie) [inline src/hooks/useActiveRoute.ts:16]: 👨‍🍳 👌

**Review by derHowie — COMMENTED** [review]
Lgtm 🙏  Curious about a specific line but otherwise no qualms. Will approve after e2e resolves.

**Comment by brunobar79** [comment]
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/@matthew/APP-1297/692862abf6495802653d23f0adddf1923b66120e.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/@matthew/APP-1297/692862abf6495802653d23f0adddf1923b66120e.ipa) for 692862abf6495802653d23f0adddf1923b66120e

File: src/components/floating-emojis/FloatingEmojis.tsx
Comment (walmat) [reply src/components/floating-emojis/FloatingEmojis.tsx:71]: this was here previous to this PR, just has my name on it since .js -> .tsx. Fine to open a ticket about it if needed

File: src/screens/SettingsSheet/utils.ts
Comment (walmat) [reply src/screens/SettingsSheet/utils.ts:48]: oops good catch ty

**Comment by brunobar79** [comment]
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/@matthew/APP-1297/8ff22715883799d59151fb49bdb899f0cd88135c.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/@matthew/APP-1297/8ff22715883799d59151fb49bdb899f0cd88135c.ipa) for 8ff22715883799d59151fb49bdb899f0cd88135c

File: src/components/floating-emojis/FloatingEmojis.tsx
Comment (derHowie) [reply src/components/floating-emojis/FloatingEmojis.tsx:71]: Oops, didn't mean to make you defend your pasta's honor. Shoulda caught it.

**Comment by brunobar79** [comment]
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/@matthew/APP-1297/9cace3bf92f65353a12b402f89be33352773974c.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/@matthew/APP-1297/9cace3bf92f65353a12b402f89be33352773974c.ipa) for 9cace3bf92f65353a12b402f89be33352773974c

File: src/screens/SettingsSheet/components/Backups/WalletsAndBackup.tsx
Comment (walmat) [reply src/screens/SettingsSheet/components/Backups/WalletsAndBackup.tsx:716]: are you opposed to this being a Zustand store?

Comment (greg-schrammel) [reply src/screens/SettingsSheet/components/Backups/WalletsAndBackup.tsx:716]: nope, a zustand store would work the same

but if we can't have a single instance of this component, it has to be a context provider that creates the store and passes it to it's children
else every place we add `<AbsolutePortalRoot />` it's gonna render all nodes in the store

Comment (walmat) [reply src/screens/SettingsSheet/components/Backups/WalletsAndBackup.tsx:716]: I see what you mean now. Okay, I'll wait to land this until the trending tokens UI work has landed so I can refactor it.

**Comment by walmat** [comment]
Blocked by https://github.com/rainbow-me/rainbow/pull/6276 due to multiple AbsolutePortal components being mounted. Merge greg's PR in first so I can rebase and refactor this to use that.

**Comment by brunobar79** [comment]
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/@matthew/APP-1297/049e61e4b219a4ba13497d75a6f1bba4ba69cacb.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/@matthew/APP-1297/049e61e4b219a4ba13497d75a6f1bba4ba69cacb.ipa) for 049e61e4b219a4ba13497d75a6f1bba4ba69cacb

**Review by BrodyHughes — APPROVED** [review]
Results of QA: https://www.notion.so/rainbowdotme/Backups-V2-Fixes-QA-Hand-off-149b001b85b4800fbf72c0f347b992b0

**Comment by brunobar79** [comment]
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/@matthew/APP-1297/ee6e027d2376278669b1065f2b9f62d88a1d1dc6.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/@matthew/APP-1297/ee6e027d2376278669b1065f2b9f62d88a1d1dc6.ipa) for ee6e027d2376278669b1065f2b9f62d88a1d1dc6

**Comment by brunobar79** [comment]
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/@matthew/APP-1297/ed377cc65b09703106f3e2df5a50f0d2e182ce13.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/@matthew/APP-1297/ed377cc65b09703106f3e2df5a50f0d2e182ce13.ipa) for ed377cc65b09703106f3e2df5a50f0d2e182ce13

File: src/components/backup/RestoreCloudStep.tsx
Comment (brunobar79) [inline src/components/backup/RestoreCloudStep.tsx:129]: If this store gets persisted (haven't checked) this is not a good idea

File: src/components/backup/RestoreCloudStep.tsx
Comment (brunobar79) [inline src/components/backup/RestoreCloudStep.tsx:145]: Same

**Review by derHowie — APPROVED** [review]
crispy stuff

referenced the notion doc to run through some flows, did not encounter any issues

File: src/components/backup/RestoreCloudStep.tsx
Comment (brunobar79) [inline src/components/backup/RestoreCloudStep.tsx:221]: I think these two lines were removed by me as part of a perf improvement and are being added back due to a merge conflict that wasn't resolved correctly

File: src/resources/transactions/consolidatedTransactions.ts
Comment (brunobar79) [inline src/resources/transactions/consolidatedTransactions.ts:138]: Aura +1000

**Review by brunobar79 — COMMENTED** [review]
LGTM! Only concern is the pwd

**Comment by brunobar79** [comment]
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/@matthew/APP-1297/7d608da67c36a508d7aa2157d171fc9d1825235a.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/@matthew/APP-1297/7d608da67c36a508d7aa2157d171fc9d1825235a.ipa) for 7d608da67c36a508d7aa2157d171fc9d1825235a

File: src/components/backup/RestoreCloudStep.tsx
Comment (walmat) [reply src/components/backup/RestoreCloudStep.tsx:129]: it is not persisted

**Review by brunobar79 — APPROVED** [review]
🚀🚀🚀

**Comment by brunobar79** [comment]
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/@matthew/APP-1297/45ba0aad4ed559b7d0055cf85cefbab775f230cb.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/@matthew/APP-1297/45ba0aad4ed559b7d0055cf85cefbab775f230cb.ipa) for 45ba0aad4ed559b7d0055cf85cefbab775f230cb

File: src/components/backup/BackupWalletPrompt.tsx
Comment (greg-schrammel) [inline src/components/backup/BackupWalletPrompt.tsx:33]:
```suggestion
  const status = backupsStore(state => state.status);
```

File: src/components/backup/ChooseBackupStep.tsx
Comment (greg-schrammel) [inline src/components/backup/ChooseBackupStep.tsx:60]:
```suggestion
  const { status, backups, mostRecentBackup } = backupsStore()
```

**Pull request merged**
